core/domain/translation_services.py

```diff
@@ -233,6 +234,57 @@ def compute_translation_related_change(
     return new_translation_models, translation_counts
 
 
+def compute_translation_related_changes_upon_revert(
+    exploration_id: str,
+    revert_to_version: int
+) -> Tuple[List[translation_models.EntityTranslationsModel], Dict[str, int]]:
+    """Create new EntityTranslation models corresponding to translation related
+    changes upon exploration revert.
+
+    Args:
+        exploration_id: str. The id of the exploration which is getting
+            reverted.
+        revert_to_version: int. The version of the exploration to which the
+            exploration is getting reverted to.
+
+    Returns:
+        Tuple(list(EntityTranslationsModel), dict(str, int)). A tuple containing
+        list of new EntityTranslationsModel and a dict with count of translated
+        contents as value and the languages as key.
+    """
+    current_exploration = exp_fetchers.get_exploration_by_id(exploration_id)
```

Comment:
Without having to fetch the exploration again, can we pass the exploration to this function and use it?

Reply:
done